theme/static/js/generic-resource.js

```diff
@@ -265,6 +265,14 @@ function update_download_status(task_id, download_path) {
     });
 }
 
+function getElementIndex(node) {
+    var index = 0;
+    while ((node = node.previousElementSibling)) {
```

Comment:
consider let

Reply:
can you use .filter and a count?

Reply:
double equals? `==`

Reply:
> can you use .filter and a count?
Not in this case, because I need a function that takes in an html element and returns its position index inside within its parent. These items could even look identical since two authors could have the same name, and thus there is not a clear way to filter.

Reply:
> double equals? `==`

The assignment operation is intentional in this case.

Reply:
> can you use .filter and a count?

After checking the code again, there is an attribute I added later which I could use to get this done through a filter function.